neo/core/container.py

```diff
@@ -73,25 +75,132 @@ def filterdata(data, targdict=None, objects=None, **kwargs):
                     results.append(obj)
-                elif (key in obj.annotations and obj.annotations[key] == value and
-                          all([obj is not res for res in results])):
+                elif (isinstance(value, FilterCondition)):
+                    if (key in obj.annotations and value.test(obj.annotations[key]) and all(
+                            [obj is not res for res in results])):
+                        results.append(obj)
+                elif key in obj.annotations and obj.annotations[key] == value and all(
+                        [obj is not res for res in results]):
                     results.append(obj)
 
     # keep only objects of the correct classes
     if objects:
         results = [result for result in results if
-                   result.__class__ in objects or
-                   result.__class__.__name__ in objects]
+                   result.__class__ in objects or result.__class__.__name__ in objects]
 
     if results and all(isinstance(obj, SpikeTrain) for obj in results):
         return SpikeTrainList(results)
     else:
         return results
 
 
+class FilterCondition():
+    """
+        FilterCondition object is given as parameter to container.filter():
+
+        example.filter(annotation=<FilterCondition>) or
+        example=filter({'annotation': <FilterCondition>})
+    """
+
+    def __init__(self, z):
+        pass
+
+    def test(self, x):
+        return True
+
+
+class FilterEqual(FilterCondition):
+
+    def __init__(self, z):
+        self.control = z
+
+    def test(self, x):
+        return x == self.control
+
+
+class FilterIsNot(FilterCondition):
+
+    def __init__(self, z):
+        self.control = z
+
+    def test(self, x):
+        return x != self.control
+
+
+class FilterLessThanEqual(FilterCondition):
+
+    def __init__(self, z):
+        self.control = z
+
+    def test(self, x):
+        return x <= self.control
+
+
+class FilterGreaterThanEqual(FilterCondition):
+    def __init__(self, z):
+        self.control = z
+
+    def test(self, x):
+        return x >= self.control
+
+
+class FilterLessThan(FilterCondition):
+
+    def __init__(self, z):
+        self.control = z
+
+    def test(self, x):
+        return x < self.control
+
+
+class FilterGreaterThan(FilterCondition):
+
+    def __init__(self, z):
+        self.control = z
+
+    def test(self, x):
+        return x > self.control
+
+
+class FilterIsIn(FilterCondition):
+
+    def __init__(self, z):
+        self.control = z
+
+    def test(self, x):
+        if (type(self.control) == list):
+            return x in self.control
+        elif (type(self.control) == int):
+            return x == self.control
+        else:
+            raise SyntaxError('parameter not of type list or int')
+
+
+class FilterInRange(FilterCondition):
+
+    def __init__(self, a, b, left_closed=False, right_closed=False):
+        if (type(a) != int or type(b) != int):
```

Comment:
can't this also be border of type `float`?

Reply:
Agreed, the border type was adjusted to `numbers`.